src/zino/statemodels.py

```diff
@@ -254,6 +261,8 @@ def set_state(self, new_state: EventState, user: str = "monitor"):
         old_state, self.state = self.state, new_state
         if (old_state, new_state) == (EventState.EMBRYONIC, EventState.OPEN):
             self.opened = now()
+        if old_state == EventState.CLOSED:
+            raise ClosedEventError
```

Comment:
I would  put the ClosedEventError check further upp, above `old_state, self.state = self.state, new_state`. It looks like `self.state` gets set to `new_state`, and then afterwards you raise the ClosedEventError. I guess the test works since this doesnt get Commited (?), but would feel cleaner to just not update `self.state` at all